cmd/podman/images/scp.go

```diff
@@ -0,0 +1,316 @@
+	imageScpCommand    = &cobra.Command{
+		Use:               "scp [options] IMAGE [IMAGE...]",
+		Long:              saveScpDescription,
+		Short:             "securely copy images",
+		RunE:              scp,
+		Args:              cobra.MinimumNArgs(1),
+		ValidArgsFunction: common.AutocompleteImages,
+		Example:           `podman image scp myimage:latest otherhost::`,
+	}
+)
+
+var (
+	scpOpts entities.ImageScpOptions
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: imageScpCommand,
+		Parent:  imageCmd,
+	})
+	scpFlags(imageScpCommand)
+}
+
+func scpFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.BoolVarP(&scpOpts.Save.Quiet, "quiet", "q", false, "Suppress the output")
+}
+
+func scp(cmd *cobra.Command, args []string) (finalErr error) {
+	var (
+		tags []string
+		err  error
+	)
+	if scpOpts.Save.Quiet { // set quiet for both load and save
+		scpOpts.Load.Quiet = true
+	}
+	f, err := ioutil.TempFile("", "podman") // open temp dir for load/save output
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(f.Name())
+
+	scpOpts.Save.Output = f.Name()
+	scpOpts.Load.Input = scpOpts.Save.Output
+	if err := parse.ValidateFileName(saveOpts.Output); err != nil {
+		return err
+	}
+	confR, err := config.NewConfig("") // create a hand made config for the remote engine since we might use remote and native at once
+	if err != nil {
+		return errors.Wrapf(err, "could not make config")
+	}
+	abiEng, err := registry.NewImageEngine(cmd, args) // abi native engine
+	if err != nil {
+		return err
+	}
+
+	cfg, err := config.ReadCustomConfig() // get ready to set ssh destination if necessary
+	if err != nil {
+		return err
+	}
+	urls, iden, serv, err := parseLocation(args, cfg) // parses connection data and "which way" we are loading and saving
+	if err != nil {
+		return err
+	}
+	// TODO: Add podman remote support
+
+	uri := []*urlP.URL{}
+	for i, url := range urls {
+		urlt, err := urlP.Parse(url) // create an actual url to pass to exec command
+		if err != nil {
+			return err
+		}
+		uri = append(uri, urlt)
+		if uri[i].User.Username() == "" {
+			if uri[i].User, err = connection.GetUserInfo(uri[i]); err != nil {
+				return err
+			}
+		}
+	}
+
+	confR.Engine = config.EngineConfig{Remote: true, CgroupManager: "cgroupfs", ServiceDestinations: serv} // pass the service dest (either remote or something else) to engine
+
+	// TODO: add podman remote support
+
+	switch {
+	case scpOpts.FromRemote: // if we want to load FROM the remote
+		_, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.ImageName, args[0], "", false)
+		if err != nil {
+			return err
+		}
+		if scpOpts.ToRemote && scpOpts.FromRemote { // we want to load remote -> remote
+			rep, err := execCommand(iden[1], uri[1], scpOpts.Save.Output, scpOpts.Load.Input, scpOpts.ImageName, scpOpts.Tag, true) // execute the command from line 207 with ToFrom defaulting to true
```

Comment:
Please don't point to lines using a line number, they get stale and confusing quickly.  As it is, I think the line number may be wrong already.